include/nbl/builtin/hlsl/bit.hlsl

```diff
@@ -0,0 +1,55 @@
+#ifndef _NBL_BUILTIN_HLSL_BIT_INCLUDED_
+#define _NBL_BUILTIN_HLSL_BIT_INCLUDED_
+
+#include <nbl/builtin/hlsl/cpp_compat.hlsl>
+
+
+#ifndef __HLSL_VERSION
+#include <bit>
+#else
+namespace nbl
+{
+namespace hlsl
+{
+
+uint32_t rotl(NBL_CONST_REF_ARG(uint32_t) x, NBL_CONST_REF_ARG(uint32_t) s)
+{
+    const uint32_t N = 32u;
+    const uint32_t r = s % N;
+    
+    if(r == 0)
+        return 0;
```

Comment:
this special case is not needed if you make the `if (r>=0)` I think in HLSL a shift by sizeof(T)*8 is likely to wipe out the bit pattern so `| (x>> (N-r))` won't be UB and just `| 0`

Reply:
done